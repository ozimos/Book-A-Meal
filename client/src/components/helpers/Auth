import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';
import { push } from 'react-router-redux';


export default function (ComposedComponent) {
  class Auth extends Component {
    componentWillMount() {
      if (!this.props.authenticated) {
        this.props.dispatch(push('/signin'));
      }
    }

    componentWillUpdate(nextProps) {
      if (!nextProps.authenticated) {
        this.props.dispatch(push('/signin'));
      }
    }
    static propTypes = {
      dispatch: PropTypes.func.isRequired,
      authenticated: PropTypes.bool.isRequired,
    }
    render() {
      return <ComposedComponent {...this.props} />;
    }
  }

  function mapStateToProps(state) {
    return { authenticated: state.loginReducer.authenticated };
  }

  return connect(mapStateToProps)(hot(module)(Auth));
}