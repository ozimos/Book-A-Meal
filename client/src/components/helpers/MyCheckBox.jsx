import { withFormsy, propTypes } from 'formsy-react';
import React from 'react';

class CheckBox extends React.Component {
  constructor(props) {
    super(props);
    this.changeValue = this.changeValue.bind(this);
  }

  changeValue(event) {

    this.props.setValue(event.target.checked);
  }

  render() {

    return (
      <React.Fragment>
        <input
          onChange={this.changeValue}
          type="checkbox"
          value={this.props.getValue() || false}
        />
      </React.Fragment>
    );
  }
}

CheckBox.propTypes = {
  ...propTypes
};
export { CheckBox };
export default withFormsy(CheckBox);
