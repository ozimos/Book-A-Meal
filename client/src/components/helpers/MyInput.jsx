import { withFormsy, propTypes } from 'formsy-react';
import PropTypes from 'prop-types';
import React from 'react';

class MyInput extends React.Component {
  constructor(props) {
    super(props);
    this.changeValue = this.changeValue.bind(this);
  }

  changeValue(event) {

    this.props.setValue(event.currentTarget.value);
  }

  render() {
    const errorMessage = this.props.getErrorMessage();

    return (
      <React.Fragment>
        <input
          style={this.props.style}
          ref={this.props.myRef}
          onChange={this.changeValue}
          type={this.props.typeOfInput}
          value={this.props.getValue() || ''}
          placeholder={this.props.placeholder}
        />
        <div
          style={{
            color: 'red', fontSize: '1rem'
          }}
        >
          {errorMessage}
        </div>
      </React.Fragment>
    );
  }
}
MyInput.defaultProps = {
  placeholder: '',
};
MyInput.propTypes = {
  typeOfInput: PropTypes.string.isRequired,
  placeholder: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  ...propTypes
};
export { MyInput };
export default withFormsy(MyInput);