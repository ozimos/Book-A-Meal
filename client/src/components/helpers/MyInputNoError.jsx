import { withFormsy, propTypes } from 'formsy-react';
import PropTypes from 'prop-types';
import React from 'react';

class MyInputNoError extends React.Component {
  constructor(props) {
    super(props);
    this.changeValue = this.changeValue.bind(this);
  }

  changeValue(event) {

    this.props.setValue(event.currentTarget.value);
  }

  render() {

    return (
      <React.Fragment>
        <input
          onChange={this.changeValue}
          type={this.props.typeOfInput}
          value={this.props.getValue() || ''}
          placeholder={this.props.placeholder}
        />
      </React.Fragment>
    );
  }
}
MyInputNoError.defaultProps = {
  placeholder: ''
};
MyInputNoError.propTypes = {
  typeOfInput: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  ...propTypes
};
export default withFormsy(MyInputNoError);