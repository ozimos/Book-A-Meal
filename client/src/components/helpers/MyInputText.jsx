import { withFormsy, propTypes } from 'formsy-react';
import PropTypes from 'prop-types';
import React from 'react';

class MyInputText extends React.Component {
  constructor(props) {
    super(props);
    this.changeValue = this.changeValue.bind(this);
  }

  changeValue(event) {
    // setValue() will set the value of the component, which in
    // turn will validate it and the rest of the form
    // Important: Don't skip this step. This pattern is required
    // for Formsy to work.
    this.props.setValue(event.currentTarget.value);
  }

  render() {
    // An error message is returned only if the component is invalid
    const errorMessage = this.props.getErrorMessage();

    return (
      <div>
        <input
          onChange={this.changeValue}
          type="text"
          value={this.props.getValue() || ''}
        />
        <span>{errorMessage}</span>
      </div>
    );
  }
}
MyInputText.propTypes = {
  setValue: PropTypes.func.isRequired,
  ...propTypes
};
export default withFormsy(MyInputText);