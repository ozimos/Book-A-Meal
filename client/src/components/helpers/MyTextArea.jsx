import { withFormsy, propTypes } from 'formsy-react';
import PropTypes from 'prop-types';
import React from 'react';

class MyTextArea extends React.Component {
  constructor(props) {
    super(props);
    this.changeValue = this.changeValue.bind(this);
  }

  changeValue(event) {

    this.props.setValue(event.currentTarget.value);
  }

  render() {
    const errorMessage = this.props.getErrorMessage();

    return (
      <React.Fragment>
        <textarea
          onChange={this.changeValue}
          value={this.props.getValue() || ''}
          placeholder={this.props.placeholder}
        />
        <div
          style={{
            color: 'red', fontSize: '1rem'
          }}
        >
          {errorMessage}
        </div>
      </React.Fragment>
    );
  }
}
MyTextArea.defaultProps = {
  placeholder: ''
};
MyTextArea.propTypes = {
  placeholder: PropTypes.string,
  ...propTypes
};
export default withFormsy(MyTextArea);