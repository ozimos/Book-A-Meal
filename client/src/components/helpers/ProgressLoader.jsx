import React from 'react';
import PropTypes from 'prop-types';

import { Row, Col, ProgressBar } from 'react-materialize';

const ProgressLoader = props =>
  (
    <div>
      <Row>
        <Col s={12}>
          <ProgressBar progress={props.upload} />
        </Col>
      </Row>
    </div>);

ProgressLoader.propTypes = {
  upload: PropTypes.number.isRequired,
};

export default ProgressLoader;

