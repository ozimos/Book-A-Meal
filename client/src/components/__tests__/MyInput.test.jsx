/* global React:false, shallow:false toJson:false */

import { MyInput } from '../helpers/MyInput';

describe('MyInput Component', () => {

  it('should render without throwing an error', () => {
    expect(shallow(<MyInput
      getValue={jest.fn()}
      getErrorMessage={jest.fn()}
      name="name"
      typeOfInput="text"
    />)
      .exists(<div
        style={{
          color: 'red', fontSize: '1rem'
        }}
      />)).toBe(true);
  });
  it('renders correctly', () => {
    const wrapper = shallow(<MyInput
      getValue={jest.fn()}
      getErrorMessage={jest.fn()}
      name="name"
      typeOfInput="text"
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

