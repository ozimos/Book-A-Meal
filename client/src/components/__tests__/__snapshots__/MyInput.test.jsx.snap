// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyInput Component renders correctly 1`] = `
<React.Fragment>
  <input
    max="99999"
    maxLength="50"
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
  <div
    style={
      Object {
        "color": "red",
        "fontSize": "1rem",
      }
    }
  />
</React.Fragment>
`;
